### PR TITLE
Maximal packet size unification

### PR DESCRIPTION
**Pull request by atiselsts**
This PR solves network stability problem reported in #799 by implementing the strategy proposed in #800.

The idea is that every radio driver should define a publicly visible constant that describes the maximal packet size (payload) supported by that driver. For 802.15.4 drivers it is 125 (the max MAC layer frame is 127 bytes, but that includes 2 bytes FCS). For example, along with `cc2538_rf_driver` this should be defined:

  #define cc2538_rf_driver_max_payload_len       125

This constant is then mapped to the name `NETSTACK_RADIO_MAX_PAYLOAD_LEN`, selected through the `NETSTACK_RADIO` define.

In `prepare` and `transmit` functions of the radio drivers now always return `RADIO_TX_ERR` code if the packet is too big to be sent.

This PR also removes the `#define SICSLOWPAN_CONF_MAC_MAX_PAYLOAD`, that was not used in the code.

This PR does not aim to improve the behaviour at the sicslowpan layer (in the scenario described in #799, the sicslowpan layer still thinks the packet size is small enough to be send without fragmentation). It just sanity checks the size at radio driver's level and makes sure to return error code to upper layers instead of failing silently.

### CODE REVIEW

File: arch/platform/jn516x/dev/micromac-radio.c [hunk 1]
```diff
@@ -463,11 +467,8 @@ prepare(const void *payload, unsigned short payload_len)
 static int
 send(const void *payload, unsigned short payload_len)
 {
-  if(prepare(payload, payload_len) == 0) {
-    return transmit(payload_len);
-  } else {
-    return RADIO_TX_ERR;
-  }
+  prepare(payload, payload_len);
```
Comment (simonduq) [inline arch/platform/jn516x/dev/micromac-radio.c]: why no longer check the return value of `prepare`?

File: arch/dev/cc1200/cc1200-conf.h [hunk 2]
```diff
@@ -72,6 +72,8 @@
 #else
 #define CC1200_MAX_PAYLOAD_LEN           127
 #endif
+
+#define cc1200_driver_max_payload_len    CC1200_MAX_PAYLOAD_LEN
```
Comment (simonduq) [inline arch/dev/cc1200/cc1200-conf.h]: Shouldn't this be 125? Won't this result in 127+2 bytes CRC frames, even without 802.15.4g?

File: arch/platform/jn516x/dev/micromac-radio.c (hunk 1 above)
Comment (atiselsts) [reply arch/platform/jn516x/dev/micromac-radio.c]: not strictly needed. but I can change it back

File: arch/dev/cc1200/cc1200-conf.h (hunk 2 above)
Comment (atiselsts) [reply arch/dev/cc1200/cc1200-conf.h]: Not sure - you're the expert on this radio! I got the impression  that 127 is ok after reading the comment above

**Comment by simonduq** [comment]
Looks good to me. Is this read?

**Pull request merged**